### PR TITLE
Harden the FreeIPA server AMI

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request makes several changes related to hardening the FreeIPA server AMI according to the CIS guidance.

## 💭 Motivation and Context

We are required to harden the OpenVPN and FreeIPA AMIs used in the COOL to obtain our ATO.

## 🧪 Testing

These changes have been used to successfully deploy a hardened FreeIPA EC2 instance in our staging COOL environment.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
HARDENED! 🧱

**Review by hillaryj — APPROVED** [review]
🦋

**Review by mcdonnnj — APPROVED** [review]
🛡️

File: src/ufw.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        - name: Allow various incoming traffic
+          ufw:
+            comment: Allow {{ item.port }} via {{ item.proto | upper }}
+            proto: "{{ item.proto }}"
+            rule: allow
+            to_port: "{{ item.port }}"
+          loop: "{{ incoming_and_outgoing_ports }}"
+        - name: Allow incoming ssh only from lo
+          ufw:
+            comment: Allow incoming ssh only from lo
```
Comment (felddy) [inline src/ufw.yml:53]: 💌 Love this 4 ever.

File: src/ufw.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    - name: Configure outgoing traffic
+      block:
+        - name: Allow various outgoing traffic
+          ufw:
+            comment: Allow {{ item.port }} via {{ item.proto | upper }}
+            direction: out
+            proto: "{{ item.proto }}"
+            rule: allow
+            to_port: "{{ item.port }}"
+          loop: "{{ incoming_and_outgoing_ports + outgoing_only_ports }}"
```
Comment (felddy) [inline src/ufw.yml:68]: Snazzy.

**Review by felddy — APPROVED** [review]
This PR has just the right amount of Zazz.  🍌

**Pull request merged**